### PR TITLE
Update default etcd server to 3.3.15 for kubernetes 1.16

### PR DESCRIPTION
**Pull request by dims**
We already bumped etcd client to 3.3.15 before code freeze but we neglected to bump the server itself. Also see additional information in https://github.com/kubernetes/kubeadm/issues/1726

Change-Id: I68f1a5e5339d83077a1a9f312c4e6e33848886c5

**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Updated default etcd service used by kubernetes to 3.3.15
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by dims** [comment]
cc @neolit123

**Comment by BenTheElder** [comment]
I assume not pushed yet?
> I0830 21:47:53.331] time="21:47:53" level=warning msg="Failed to pull k8s.gcr.io/etcd:3.3.15 with error: exit status 1"

**Comment by dims** [comment]
@BenTheElder correct, we need to build and push that

**Comment by dims** [comment]
/assign @jpbetz @yuwenma 
cc @gyuho

**Comment by dims** [comment]
i am now able to run `make build test` from cluster/images/etcd directory. looking at the README, `make all-build` works as well. Once we get reviews, we need to run `make all-push` to push the new images to production.

/assign @liggitt 

(for reviewing the image version/tag etc)

**Comment by dims** [comment]
/hold

for review and for publishing images

**Comment by neolit123** [comment]
thanks @dims
xref https://github.com/kubernetes/kubeadm/issues/1726

**Comment by dims** [comment]
cc @kubernetes/release-managers

**Comment by justaugustus** [comment]
For Release Managers:
/lgtm
/approve
/retest

File: cluster/images/etcd/migrate/integration_test.go
Comment (cblecker) [inline cluster/images/etcd/migrate/integration_test.go]: This is new.. I realize this is only a test, but is an init function the best way to handle this?

Comment (dims) [reply cluster/images/etcd/migrate/integration_test.go]: we already have this pattern @cblecker : 
https://cs.k8s.io/?q=logtostderr.*true&i=nope&files=%5E.*%5C_test%5C.go&repos=kubernetes/kubernetes

Currently the integration test fails (even before this PR i think), this one for the record:
https://github.com/kubernetes/kubernetes/blob/master/cluster/images/etcd/Makefile#L177-L178

**Comment by neolit123** [comment]
@dims so i guess to make the tests pass, the image has to be pushed before this PR merges?
i'm wondering whether we should bump first and then modify framework, kubeadm, once the image is up at GCR.

- update image build to add 3.3.15
- upload image at GCR
- update the image usage

**Comment by dims** [comment]
@neolit123 i'd like to iterate on this PR itself (just like last time). We can experiment next time earlier in the cycle hopefully.

**Comment by jpbetz** [comment]
Sorry for the belated response, I was on vacay. The change looks right to me. I don't see the published image in the repo, so I'll kick that process off now.

**Comment by jpbetz** [comment]
Staging images published and image promotion request created.

**Comment by justaugustus** [comment]
@jpbetz -- can you link the promotion request here?

**Review by timothysc — CHANGES_REQUESTED** [review]
You'll need to re-vendor the client update too. 
/assign @timothysc

**Comment by jpbetz** [comment]
> @jpbetz -- can you link the promotion request here?

There is a google internal process. I don't think there is a way to link to it. But the images are now published.

**Comment by neolit123** [comment]
thanks @jpbetz 
looks like the image tag is `3.3.15-0`, so it has to be like that in this PR:
https://console.cloud.google.com/gcr/images/google-containers/GLOBAL/etcd?gcrImageListsize=30

**Comment by neolit123** [comment]
@timothysc 
> You'll need to re-vendor the client update too.

i think we already did that?

**Review by timothysc — APPROVED** [review]
/approve

**Comment by neolit123** [comment]
/test pull-kubernetes-e2e-kind
(this should fail the first run)

**Comment by neolit123** [comment]
> time="18:18:43" level=info msg="Failed to pull image: k8s.gcr.io/etcd:3.3.15" error="exit status 1"

@dims the image tag should be `3.3.15-0`.
hopefully the revision doesn't break something.

**Comment by dims** [comment]
@neolit123 Hmmmm

**Comment by dims** [comment]
@jpbetz there’s a problem with the image… compare

- https://console.cloud.google.com/gcr/images/google-containers/GLOBAL/etcd@sha256:12c2c5e5731c3bcd56e6f1c05c0f9198b6f06793fa7fca2fb43aab9622dc4afa/details?tab=info
- https://console.cloud.google.com/gcr/images/google-containers/GLOBAL/etcd@sha256:17da501f5d2a675be46040422a27b7cc21b8a43895ac998b171db1c346f361f7/details?tab=info

the second one has 3.3.10 and 3.3.10-0
the first one just has 3.3.15-0

**Comment by neolit123** [comment]
haven't looked at the layers. i tried `docker pull gcr.io/google-containers/etcd:3.3.15-0` and it work.
`docker pull gcr.io/google-containers/etcd:3.3.15` doesn't and that is the reason for the `pull-kubernetes-e2e-kind ` failure at least.

**Comment by jpbetz** [comment]
3.3.15 image tags published

/retest

**Comment by neolit123** [comment]
cc @liggitt 
PTAL for approval of the `staging/src/k8s.io/sample-apiserver/`
`/cluster` too i guess.

**Comment by neolit123** [comment]
thanks
/hold cancel

**Comment by dims** [comment]
node-e2e is not flaky, it's missing an image - fix is here https://github.com/kubernetes/test-infra/pull/14203

**Pull request merged**
